### DISCUSSION TITLE
How to use addChild and setParent?

### DISCUSSION THREAD

**Discussion by lastlos** [osch/lua-lwtk#7] [Q&A] [answered]
My first question: I would like to create childs (If I'm not mistaken, copies of widget) of a widget but i dont know how to do it. I didn't find how to use it. I tried create child with setting parent but It didn't work.
My second question: Is Lwtk got a manuals or tutorials? If it have, where?

**Comment by osch** [comment, marked as answer]
Here is a simple example using `addChild`:
```lua
local lwtk = require("lwtk")

local Application    = lwtk.Application
local Column         = lwtk.Column
local Row            = lwtk.Row
local PushButton     = lwtk.PushButton
local TitleText      = lwtk.TitleText
local TextLabel      = lwtk.TextLabel
local Space          = lwtk.Space

local app = Application("example")

local win = app:newWindow {
    title = "example",
    Column {
        id = "column1",
        TitleText  { text = "Hello World!", style = { textSize = 35 } },
        Row {
            Space {},
            PushButton { id = "button1",
                         text = "&Add Child" },
            Space {}
        }
    },
}

local childCount = 0

win:childById("button1"):setOnClicked(
    function(widget)
        local column = widget:byId("column1")
        childCount = childCount + 1
        column:addChild(TextLabel{
            text = "Child "..childCount,
            style = { TextAlign = "center" }
        })
    end)

win:show()
app:runEventLoop()
```

There is no method `setParent`. The method `_setParent` begins with an underline to indicate that it should not be used in normal cases.

> Is Lwtk got a manuals or tutorials? If it have, where?

lwtk's [documentation](https://github.com/osch/lua-lwtk/blob/master/doc/README.md) is work in progress and not completed yet, you can also study the [examples](https://github.com/osch/lua-lwtk/blob/master/example/README.md#lwtk-examples).

**Reply by lastlos** [reply]
What If children doesn't fit to window? Is there is (lwtk) a scroll bar for it?

**Reply by lastlos** [reply]
and how to clear children?

**Reply by osch** [reply]
> how to clear children

You might want to take a look at the latest commit https://github.com/osch/lua-lwtk/commit/db8f16c17e7772390ff9018ac178368bac417e71 that implements a `removeChild` method.
